contrib/database/sql/conn.go

```diff
@@ -203,5 +203,8 @@ func (tp *traceParams) tryTrace(ctx context.Context, qtype queryType, query stri
 			span.SetTag(k, v)
 		}
 	}
-	span.Finish(tracer.WithError(err))
+	if err != nil && tp.cfg.errCheck != nil && tp.cfg.errCheck(err) {
+		span.SetTag(ext.Error, err)
+	}
+	span.Finish()
```

Comment:
What happens when `err != nil` and `tp.cfg.errCheck == nil`? Are we not going to report the error? A test should be failing.

Reply:
We should probably be consistent with https://cs.github.com/DataDog/dd-trace-go/blob/3528a0d641097d68b3b9e750a51bd8f79c72d360/contrib/gopkg.in/jinzhu/gorm.v1/gorm.go#L137. If we aren't doing that check here, why is this case different?
I actually think having a reasonable default here would be better: https://cs.github.com/DataDog/dd-trace-go/blob/3528a0d641097d68b3b9e750a51bd8f79c72d360/contrib/jinzhu/gorm/option.go#L35
Then we don't have to worry about the nil case, unless there is a way for someone to override the default and set it to nil?

Reply:
@katiehockman We discussed about it here. https://github.com/DataDog/dd-trace-go/pull/1315#discussion_r887712301
how about you ?

Reply:
I agree with Katie. Can we do the same here?

Reply:
@gbbr @katiehockman I think it's difficult to be same with gorm option.
Because gorm options are passed from only one point. So no problem to set default function variable in `defaults`. When user pass errcheck option, just overwrided.

But `contrib/database/sql` has 2 points to pass options. Register and OpenDB.
And In OpenDB, if config is empty, option value are came from set in RegisterDB.

Like that.

```go
	rc, _ := registeredDrivers.config(name)
	cfg := new(config)
	defaults(cfg)
	for _, fn := range opts {
		fn(cfg)
	}
	// use registered config for unset options
	if cfg.serviceName == "" {
		cfg.serviceName = rc.serviceName
	}
	if math.IsNaN(cfg.analyticsRate) {
		cfg.analyticsRate = rc.analyticsRate
	}
```

It mean we need to detect cfc.errcheck has default value at here.
But if we set default function variable to errcheck in defaults, we cant know cfc.errcheck is default value or not.
So I introduce to set nil value as default for cfc.errcheck.

Reply:
Makes sense. Thanks for the explanation.